sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample2_ExecuteSparkStatement.md

```diff
@@ -0,0 +1,200 @@
+SparkSessionClient client = new SparkSessionClient(new Uri(endpoint), sparkPoolName, new DefaultAzureCredential());
+```
+
+## Create Spark session
+
+All `SparkStatement` are created within the context of a `SparkSession`, so first create a spark session passing in details on the requested host.
+
+```C# Snippet:CreateSparkSession
+SparkSessionOptions request = new SparkSessionOptions(name: $"session-{Guid.NewGuid()}")
+{
+    DriverMemory = "28g",
+    DriverCores = 4,
+    ExecutorMemory = "28g",
+    ExecutorCores = 4,
+    ExecutorCount = 2
+};
+
+SparkSession sessionCreated = client.CreateSparkSession(request);
+
+// Waiting session creation completion
+sessionCreated = PollSparkSession(client, sessionCreated);
+```
+
+## Retrieve a Spark Session
+
+To retrieve an existing session call `GetSparkSession`, passing in the session ID.
+
+```C# Snippet:GetSparkSession
+SparkSession session = client.GetSparkSession(sessionCreated.Id);
+Debug.WriteLine($"Session is returned with name {session.Name} and state {session.State}");
+```
+
+## Creating Spark statements
+
+To create statements within a session call `CreateSparkStatement`, passing in both the statements details in a `SparkStatementOptions` along with the ID of the session.
+
+To wait for the statement's completion a `PollSparkStatement` call is required. This support code will be detailed below, and it hoped to be temporary. 
+
+```C# Snippet:CreateSparkStatement
+SparkStatementOptions sparkStatementRequest = new SparkStatementOptions
+{
+    Kind = SparkStatementLanguageType.Spark,
+    Code = @"print(""Hello world\n"")"
+};
+SparkStatement statementCreated = client.CreateSparkStatement(sessionCreated.Id, sparkStatementRequest);
+
+// Wait operation completion
+statementCreated = PollSparkStatement(client, sessionCreated.Id, statementCreated);
+```
+
+## Retrieve a statement
+
+To retrieve an existing statement call `GetSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:GetSparkStatement
+SparkStatement statement = client.GetSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is returned with id {statement.Id} and state {statement.State}");
+```
+
+## Cancel a statement
+
+To cancel a submitted statement call `CancelSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:CancelSparkStatement
+SparkStatementCancellationResult cancellationResult = client.CancelSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is cancelled with message {cancellationResult.Msg}");
+```
+
+## Cancel a session
+
+To cancel the entire Spark session  call `CancelSparkSession`, passing in the session ID.
+
+```C# Snippet:CancelSparkSession
+Response operation = client.CancelSparkSession(sessionCreated.Id);
+```
+
+## Support Code
+
+Today the following support code is needed to poll the status of submitted Spark statements and sessions.
```

Comment:
You may want to consider collapsing these into a single paragraph.   It may also be helpful to offer a bit of detail about which part is temporary and how it may evolve, without going into specifics.